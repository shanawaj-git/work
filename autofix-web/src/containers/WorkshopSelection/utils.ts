import {
  BoundPayloadType,
  GetDirections,
  LatLngType,
  MapObjType,
} from "./interfaces";

export const markerIcon = (icon: string) => ({
  scaledSize: {
    height: 24,
    width: 24,
  },
  url: icon,
});

export const getBoundLocation = ({
  map,
  onBoundChange,
}: {
  map: MapObjType;
  onBoundChange: (e: BoundPayloadType) => void;
}) => {
  const bounds = map.getBounds();
  const southWest = getSouthWestCoordinates(bounds);
  const northEast = getNorthEastCoordinates(bounds);
  const result = buildBoundPayload(northEast, southWest);
  onBoundChange(result);
};

const getSouthWestCoordinates = (bounds: GetDirections): LatLngType => ({
  lat: bounds.getSouthWest().lat(),
  lng: bounds.getSouthWest().lng(),
});

const getNorthEastCoordinates = (bounds: GetDirections): LatLngType => ({
  lat: bounds.getNorthEast().lat(),
  lng: bounds.getNorthEast().lng(),
});

const buildBoundPayload = (
  northEast: LatLngType,
  southWest: LatLngType
): BoundPayloadType => ({
  locationBoxBounds: {
    bottomLeft: {
      type: "Point",
      coordinates: [southWest.lng, southWest.lat],
    },
    topRight: { type: "Point", coordinates: [northEast.lng, northEast.lat] },
  },
});
