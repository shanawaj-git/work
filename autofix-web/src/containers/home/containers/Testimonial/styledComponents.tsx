import styled from "styled-components";
import { Carousel } from "@albathanext/design-system";
import { utils } from "@albathanext/design-system";

export const StyledCarousel = styled(Carousel)`
  padding-right: 24px;
  padding-left: 24px;
  margin-top: 40px;
  flex: 1;
  display: flex;
  flex-direction: column;

  .MuiMobileStepper-dot,
  .MuiMobileStepper-dot {
    border-radius: 0;
    width: 9.59px;
    height: 1px;
    opacity: 0.4;
    border: 1px solid rgba(255, 255, 255, 0.8);
  }

  .MuiMobileStepper-dotActive {
    width: 88px;
    border: 1px solid ${utils.colorSelector("e100")};
  }

  .MuiButton-root {
    min-width: 36px;
    min-height: 36px;
    width: 36px;
    height: 36px;
    border-radius: 2px;
    background: ${utils.colorSelector("p300")};
    border: 1px solid ${utils.colorSelector("b100")};
    margin-right: 0;
  }

  .MuiButton-root[disabled] {
    border: 1px solid ${utils.colorSelector("b100")};
    background: linear-gradient(0deg, #1f2222, #1f2222),
      linear-gradient(0deg, rgba(220, 220, 220, 0.2), rgba(220, 220, 220, 0.2));
  }

  .MuiPaper-root {
    background: transparent;
    padding-left: 0;
    padding-right: 0;
  }
`;
