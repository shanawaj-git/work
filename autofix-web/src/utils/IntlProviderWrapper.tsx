import React, { useMemo } from "react";
import { getTranslation, TranslationsType } from "../translations";
import { IntlProvider } from "react-intl";
import { flatData, I18NLocale } from ".";

export const withWrapper = (component: JSX.Element, pageProps = { en: {} }) => (
  <IntlProviderWrapper locale="en" pageProps={pageProps}>
    {component}
  </IntlProviderWrapper>
);

export const IntlProviderWrapper = ({
  children,
  pageProps,
  locale,
}: {
  children: JSX.Element;
  pageProps: TranslationsType;
  locale: I18NLocale;
}) => {
  const messages = useMemo(() => {
    return {
      ...getTranslation(locale),
      ...flatData(pageProps[locale] || {}),
    };
  }, [locale, pageProps]);

  return (
    <IntlProvider locale={`${locale}`} messages={messages}>
      {children}
    </IntlProvider>
  );
};
