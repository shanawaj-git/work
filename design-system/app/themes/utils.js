import { getTheme } from './getTheme';

const BOX_SHADOW_MAPPING = {
  level1: '0px 3px 10px rgba(0, 0, 0, 0.07)',
  level2: '0px 6px 10px rgba(0, 0, 0, 0.07)',
  level3: '0px 10px 20px rgba(0, 0, 0,  0.07)',
  level4: '0px 14px 28px rgba(0, 0, 0, 0.1)',
  level5: '0px 19px 38px rgba(0, 0, 0, 0.15)',
  focus: `0px 0px 10px ${getTheme().theme.color.d100}`,
};

const boxShadowSelector = level => BOX_SHADOW_MAPPING[level];
const colorSelector = color => getTheme().theme.color[color.toLowerCase()];
export { boxShadowSelector, colorSelector };
