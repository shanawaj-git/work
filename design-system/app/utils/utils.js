import { v4 } from 'uuid';
import _debounce from 'lodash/debounce';
/**
 * Helper: Convert Array of coordinates to an object of cordinates
 * @param {*} cordinates : Array [lat,lng]
 * @returns
 */
export const transformedCordinates = cordinates => {
  const [lat, lng] = cordinates;
  return {
    lat,
    lng,
  };
};

export const readableCordinates = ({ lat, lng }) => ({
  lat: lat(),
  lng: lng(),
});

export const uuid = () => v4();

const MAP_ADDRESS_COMPONENT_TYPE = {
  LOCALITY: 'locality',
  COUNTRY: 'country',
};

export const getCityAndCountry = address => {
  let city;
  let country;
  // eslint-disable-next-line no-unused-expressions
  address?.forEach(component => {
    const { types } = component;
    if (types.indexOf(MAP_ADDRESS_COMPONENT_TYPE.LOCALITY) > -1) {
      city = component.long_name;
    }

    if (types.indexOf(MAP_ADDRESS_COMPONENT_TYPE.COUNTRY) > -1) {
      country = component.long_name;
    }
  });
  return {
    city,
    country,
  };
};

export const debounce = (func, wait) => _debounce(func, wait);

export const canBeRenderedInTypographyComponent = obj =>
  typeof obj === 'string' ||
  obj?.type?.displayName === 'MemoizedFormattedMessage';
