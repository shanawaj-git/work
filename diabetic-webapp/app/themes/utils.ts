
import { diabeticTheme } from './theme';

const BOX_SHADOW_MAPPING = {
  level1: '0px 3px 10px rgba(0, 0, 0, 0.07)',
  level2: '0px 6px 10px rgba(0, 0, 0, 0.07)',
  level3: '0px 10px 20px rgba(0, 0, 0,  0.07)',
  level4: '0px 14px 28px rgba(0, 0, 0, 0.1)',
  level5: '0px 19px 38px rgba(0, 0, 0, 0.15)',
};

const boxShadowSelector = (level: string): string => BOX_SHADOW_MAPPING[level];

const colorSelector = (color: string): string =>
  diabeticTheme.color[color.toLowerCase()];

export { boxShadowSelector, colorSelector };
