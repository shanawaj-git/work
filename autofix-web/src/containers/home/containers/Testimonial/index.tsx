import { NextPage } from "next";
import Section from "@/components/Section";
import { Article, utils } from "@albathanext/design-system";
import { StyledCarousel } from "@/containers/home/containers/Testimonial/styledComponents";
import { FormattedMessage } from "react-intl";
import React, { useEffect, useRef, useState } from "react";
import { StyledTestimonialCard } from "@/components/TestimonialCard/styledComponents";
import { TestimonialType } from "@/services/home/testimonial";

interface Shape {
  offsetHeight: number;
}

const Testimonial: NextPage<{ testimonials: TestimonialType[] }> = ({
  testimonials,
}) => {
  const [sliderContainerHeight, setSlideContainerHeight] = useState(0);

  const carouselRef: React.MutableRefObject<Shape | undefined> = useRef();

  useEffect(() => {
    if (!carouselRef.current || sliderContainerHeight !== 0) return;
    const height: number = carouselRef.current.offsetHeight;
    setSlideContainerHeight(height);
  }, [carouselRef.current]);

  function content(activeStep: number, height: number) {
    return testimonials?.map((step, index) => (
      <div key={index}>
        {Math.abs(activeStep - index) <= 2 ? (
          <StyledTestimonialCard
            testId={`testimonial-card-${index}`}
            className={"testimonial-card"}
            content={step}
            height={height}
          />
        ) : null}
      </div>
    ));
  }

  return (
    <Section
      data-testid="testimonial-section"
      style={{
        background: utils.colorSelector("p600"),
      }}
      className="flex flex-col justify-between py-[100px] text-white bg-cover bg-no-repeat"
    >
      <div className={"flex flex-col flex-1 "}>
        <div className="px-6">
          <Article
            title={<FormattedMessage id="home.testimonial.title" />}
            divider
            dividerProps={{
              style: { borderColor: utils.colorSelector("P300") },
              className: "mt-2",
            }}
            titleProps={{
              color: "white",
              className: "text-4xl whitespace-pre-line",
            }}
          />
        </div>
        <StyledCarousel
          _ref={carouselRef}
          className={"carousel h-full"}
          direction="ltr"
          nextButton={true}
          backButton={false}
          content={content(1, sliderContainerHeight)}
          maxSteps={testimonials?.length || 0}
        />
      </div>
    </Section>
  );
};

export default Testimonial;
