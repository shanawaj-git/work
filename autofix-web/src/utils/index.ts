import { itemType } from "@/containers/RecommendedMaintenance";
import {
  AssetType,
  fetchCommonData,
  FooterType,
  getAssets,
  getAssetsData,
  getLabelData,
  getLabels,
  HomeActionType,
  LabelType,
  ServiceType,
} from "@/services/common";
import { TestimonialType } from "@/services/home/testimonial";

export const utils = "UTILS";
const GRAPHQL_CLIENT_READ_TIMEOUT = 5000;

interface getCommonPageContentParams {
  locale: I18NLocale;
  pageName: string;
}

export interface HomeContentType {
  labels?: LabelType[];
  assets?: AssetType[];
  services?: ServiceType[];
  actions?: HomeActionType[];
  testimonials?: TestimonialType[];
  footer?: FooterType;
}

export type I18NLocale = "en" | "ar";

/**
 * Will Return a Data object with Labels and Assets Regarding the page from Strapi
 * @param  locale , pageName
 * @returns Object with Labels and Assets
 */

export const getCommonPageContent = async ({
  locale,
  pageName,
}: getCommonPageContentParams): Promise<HomeContentType> => {
  let data: HomeContentType = {};
  return new Promise(async (resolve, reject) => {
    try {
      const commonData = await (
        await fetchCommonData(locale as I18NLocale, pageName)
      ).data;
      const assets = getAssets(commonData);
      const assetsData = getAssetsData(assets);
      const labels = getLabelData(getLabels(commonData));
      data = { assets: assetsData, labels };
      resolve(data);
    } catch (error) {
      resolve(data);
    }
  });
};

interface rejectEarlyType<T> {
  fn: () => Promise<T>;
  def: T;
  timeout?: number;
}

export const rejectEarly = <T>({
  fn,
  def,
  timeout = GRAPHQL_CLIENT_READ_TIMEOUT,
}: rejectEarlyType<T>): Promise<T> => {
  return new Promise(async (resolve, _reject) => {
    setTimeout(() => {
      resolve(def);
    }, timeout);
    try {
      const out: T = await fn();
      resolve(out);
    } catch (error) {
      resolve(def);
    }
  });
};

export const validateEmail = (email: string) =>
  /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(email);

export const flatData = (obj: { [key: string]: {} }) =>
  Object.values(obj).reduce((acc: {}, curr: {}) => {
    return { ...acc, ...curr };
  }, {});

export const buildImageUrl = (path: string): string =>
  process.env.NEXT_PUBLIC_STRAPI_SERVICE?.concat(path) || "";

export const groupBy = <T>(
  array: T[],
  predicate: (value: T, index: number, array: T[]) => string
) =>
  array.reduce((acc, value, index, array) => {
    (acc[predicate(value, index, array)] ||= []).push(value);
    return acc;
  }, {} as { [key: string]: T[] });

export const includesCaseInsensitive = (
  string: string,
  substring: string
): Boolean => {
  return string.toLowerCase().includes(substring.toLowerCase());
};
export const getTotalForMax = (data: itemType[]) => {
  return data.reduce((acc: number, curr: { cost: string }) => {
    let value = acc;
    value += +parseFloat(
      (curr.cost.match(/[0-9]+/gm) || ["0"]).splice(-1)[0]
    ).toFixed(2);
    return value;
  }, 0);
};

export const getCurrency = (data: itemType[]) => {
  return (data[0].cost.match(/[A-Z]+/gm) || ["USD"]).splice(-1)[0];
};

export const getCurrentLocation = () => {
  return new Promise((resolve: (e: GeolocationCoordinates) => void, reject) => {
    if (!navigator) throw new Error("Navigator doesn't exist");
    if (!navigator.geolocation) throw new Error("Geolocation doesn't exist");
    navigator.geolocation.getCurrentPosition(
      (position: GeolocationPosition) => {
        return resolve(position.coords);
      },
      reject,
      {
        maximumAge: 5000,
        enableHighAccuracy: true,
      }
    );
  });
};

interface getCurrentLiveLocationType {
  callBack: (e: GeolocationPosition) => void;
  onError: (e: GeolocationPositionError | Error) => void;
}

export const getCurrentLiveLocation = ({
  callBack,
  onError,
}: getCurrentLiveLocationType) => {
  if (!navigator) throw new Error("Navigator doesn't exist ");
  if (!navigator.geolocation) throw new Error("Geolocation doesn't exist ");
  navigator.geolocation.watchPosition((position: GeolocationPosition) => {
    callBack(position);
  }, onError);
};
