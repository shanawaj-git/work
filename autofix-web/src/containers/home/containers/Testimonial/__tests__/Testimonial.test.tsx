import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { withWrapper } from "@/utils/IntlProviderWrapper";
import Testimonial from "../index";

describe("<Testimonial />", () => {
  const testimonials = [
    {
      title: "Testimonial",
      content: "This is a testimonial from a customer",
      img: "https://s3-alpha-sig.figma.com/img/8458/48c5/b622b006cfb8fafc6bd63b7c877969e3?Expires=1659916800&Signature=UUriV5UFOKrlOd1K2sb4AjJkrAJmCs79jNMSAfj2zStiTYJuzJrcRkSlzA69xu7D8xqrdSFqe339NMwJHlS6L5ZXYqv64tm2s0vQk0Gqczc12FQUk~6W5je1l6loLG289IYoHVdMnVwzJXO-rG7lqu8sPeSm7-I~AkeFwqF3uDjese-I--7q~6n10SsfeEGWSTPbpn4eaeL7kd-AzhBKKMpl8NvAkNu5kncmL4nGBwy-8f-wW7ve5r~R1ZlOAP1G89CRzanxOUOk84G1-28HBIit33gzmIPYTD0Pz31s1Rwqi9viB9kF~c6G~N5TulZ8AKmMIZl92pW6IcWVuctMiQ__&Key-Pair-Id=APKAINTVSUGEWH5XD5UA",
      rating: 3,
      customer: "Customer ",
    },
    {
      title: "Testimonial",
      content: "This is a another testimonial from a customer",
      img: "https://s3-alpha-sig.figma.com/img/8458/48c5/b622b006cfb8fafc6bd63b7c877969e3?Expires=1659916800&Signature=UUriV5UFOKrlOd1K2sb4AjJkrAJmCs79jNMSAfj2zStiTYJuzJrcRkSlzA69xu7D8xqrdSFqe339NMwJHlS6L5ZXYqv64tm2s0vQk0Gqczc12FQUk~6W5je1l6loLG289IYoHVdMnVwzJXO-rG7lqu8sPeSm7-I~AkeFwqF3uDjese-I--7q~6n10SsfeEGWSTPbpn4eaeL7kd-AzhBKKMpl8NvAkNu5kncmL4nGBwy-8f-wW7ve5r~R1ZlOAP1G89CRzanxOUOk84G1-28HBIit33gzmIPYTD0Pz31s1Rwqi9viB9kF~c6G~N5TulZ8AKmMIZl92pW6IcWVuctMiQ__&Key-Pair-Id=APKAINTVSUGEWH5XD5UA",
      rating: 5,
      customer: "Customer ",
    },
  ];

  test("it should render component with all testimonials", () => {
    const component = render(
      withWrapper(<Testimonial testimonials={testimonials} />)
    );

    const firstTestimonialCard = component.getByTestId(`testimonial-card-0`);

    expect(firstTestimonialCard).not.toBeNull();
  });

  test("it should render component and switch testimonial card when button clicked", async () => {
    const component = render(
      withWrapper(<Testimonial testimonials={testimonials} />)
    );

    const nextButton = component.getByRole("button");

    await nextButton.click();
    const secondTestimonialCard = component.getByTestId(`testimonial-card-1`);
    expect(secondTestimonialCard).not.toBeNull();
  });

  test("it should render component and disable next button when it is last card", async () => {
    const component = render(
      withWrapper(<Testimonial testimonials={testimonials} />)
    );
    const nextButton = component.getByRole("button");

    await nextButton.click();

    const disabledNextButton = component.getByRole("button");
    expect(disabledNextButton).toHaveProperty("disabled", true);
  });

  test("it should render empty component when no testimonials available", () => {
    const component = render(withWrapper(<Testimonial testimonials={[]} />));

    expect(() => component.getByTestId("testimonial-card-0")).toThrow();
  });

  test("it should render component and match snapshot", () => {
    const { container } = render(
      withWrapper(<Testimonial testimonials={testimonials} />)
    );

    expect(container.innerHTML).toMatchSnapshot();
  });
});
