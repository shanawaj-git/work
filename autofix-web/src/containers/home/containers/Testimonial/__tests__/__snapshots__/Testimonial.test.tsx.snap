// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Testimonial /> it should render component and match snapshot 1`] = `
"<div data-testid=\\"testimonial-section\\" class=\\"min-h-screen flex flex-col justify-between py-[100px] text-white bg-cover bg-no-repeat\\"><div class=\\"flex flex-col flex-1 \\"><div class=\\"px-6\\"><div data-testid=\\"article-component\\"><p class=\\"MuiTypography-root MuiTypography-body1 text-4xl whitespace-pre-line css-x61q1w-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> Trusted
by customers</span></p><div data-testid=\\"article-component-divider\\" style=\\"border-color: #e25f6d;\\" class=\\"w-10 h-0 border-solid border-2 my-4 border-[#E25F6D] mt-2\\"></div><div color=\\"white\\" class=\\"text-4xl whitespace-pre-line\\"></div></div></div><div class=\\"sc-d6b1016c-0 jvznlT carousel h-full\\"><div style=\\"overflow-x: hidden; flex: 1; flex-direction: flex; height: 100%;\\"><div style=\\"flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;\\" class=\\"react-swipeable-view-container\\"><div style=\\"width: 100%; flex-shrink: 0; overflow: auto;\\" aria-hidden=\\"false\\" data-swipeable=\\"true\\"><div><div class=\\"sc-f709edfe-0 ixOjLN testimonial-card\\" data-testid=\\"testimonial-card-0\\"><div class=\\"testimonial-card-text\\" data-testid=\\"testimonial-card-content-testimonial-card-0\\">This is a testimonial from a customer</div><div class=\\"testimonial-card-footer\\"><div class=\\"testimonial-card-stars\\" data-testid=\\"testimonial-card-stars-testimonial-card-0\\"><img alt=\\"start-0\\" src=\\"star.svg\\" class=\\"testimonial-card-star\\"><img alt=\\"start-1\\" src=\\"star.svg\\" class=\\"testimonial-card-star\\"><img alt=\\"start-2\\" src=\\"star.svg\\" class=\\"testimonial-card-star\\"></div><p class=\\"testimonial-card-customer\\" data-testid=\\"testimonial-card-customer-testimonial-card-0\\">Customer </p></div></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiMobileStepper-root MuiMobileStepper-positionStatic css-rh92k-MuiPaper-root-MuiMobileStepper-root\\"><div class=\\"MuiMobileStepper-dots css-1384nzf-MuiMobileStepper-dots\\"><div class=\\"MuiMobileStepper-dot MuiMobileStepper-dotActive css-26w9jf-MuiMobileStepper-dot\\"></div><div class=\\"MuiMobileStepper-dot css-114p2tk-MuiMobileStepper-dot\\"></div></div><button class=\\"root StyledButton-sc-1apep4b-0 bXswYh MuiButton-root text MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"next-btn\\"><div style=\\"visibility: visible;\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"ArrowForwardIcon\\"><path d=\\"m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z\\"></path></svg></div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div></div></div>"
`;
